src/core/chunked_stream.js

```diff
@@ -98,6 +98,10 @@ class ChunkedStream {
   }
 
   ensureByte(pos) {
+    if (pos < this.progressiveDataLength) {
```

Comment:
I assume `pos` is 0-indexed, so the `<` check will work. However, why are we using `<=` then in `ensureRange` below? I think the `end` is also a 0-indexed position, so it that wrong below?

Reply:
The way that I've understood `MissingDataException`, and related code, is that `begin` is inclusive while `end` is *exclusive*, which should thus explain things; note the formatting used in https://github.com/mozilla/pdf.js/blob/47f208df360c750c40f06d4573377fd6e90b6e6d/src/core/core_utils.js#L32-L35 and compare with standard interval notation as outlined in https://en.wikipedia.org/wiki/Interval_(mathematics)#Including_or_excluding_endpoints

To be absolutely sure though, I suppose that you might want @yurydelendik to weigh in here!

Reply:
@yurydelendik Sorry to bother you, but any chance that you have time to comment here?

Reply:
Also, given that a `ensureByte(pos)` call should essentially be equal to a `ensureRange(pos, pos + 1)` call, that would mean a `if (pos + 1 <= this.progressiveDataLength) {` check in the latter case which is how I arrived at the condition under discussion here.